agr_literature_service/api/crud/referencefile_crud.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from fastapi import HTTPException, status
+from fastapi.encoders import jsonable_encoder
+from sqlalchemy.orm import Session
+
+from agr_literature_service.api.crud.referencefile_utils import read_referencefile_db_obj_from_md5sum_or_id
+from agr_literature_service.api.models import ReferencefileModel, ReferenceModel
+from agr_literature_service.api.schemas.referencefile_mod_schemas import ReferencefileModSchemaPost
+from agr_literature_service.api.schemas.referencefile_schemas import ReferencefileSchemaPost
+from agr_literature_service.api.schemas.response_message_schemas import messageEnum
+from agr_literature_service.api.crud.referencefile_mod_crud import create as create_referencefile_mod
+
+logger = logging.getLogger(__name__)
+
+
+def create(db: Session, request: ReferencefileSchemaPost):
+    request_dict = request.dict()
+    ref_obj = db.query(ReferenceModel).filter(ReferenceModel.curie == request.reference_curie).one_or_none()
+    if ref_obj is None:
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                            detail=f"Reference with curie {request.reference_curie} does not exist")
+    del request_dict["reference_curie"]
+    request_dict["reference_id"] = ref_obj.reference_id
+    mod_abbreviation = request_dict["mod_abbreviation"]
+    del request_dict["mod_abbreviation"]
+    new_ref_file_obj = ReferencefileModel(**request_dict)
+    db.add(new_ref_file_obj)
+    db.commit()
+    if mod_abbreviation is not None:
```

Comment:
This isn't right, is it ?  We want to be able to create a referencefile with a null mod, which would create a referencefile_mod with a null mod, meaning it's PMC ?  Or for the PMC case would we require creating the referencefile first, then explicitly creating the referencefile_mod with a null mod ?   
Same issue with the referencefile_mod create.  
We should probably add a test for that.

Reply:
fixed by allowing null values for PMC